### PR TITLE
Add fields to info metricset of Redis module

### PR DESCRIPTION
**Pull request by a3dho3yn**
Memory fragmentation ratio, active defragmentation stats, slave status and some other fields was missing from `info` metricset.

### CODE REVIEW

File: metricbeat/module/redis/info/data.go [hunk 1]
```diff
@@ -84,25 +108,59 @@ var (
...
+					"left_bytes":          c.Int("master_sync_left_bytes", s.Optional),
+					"last_io_seconds_ago": c.Int("master_sync_last_io_seconds_ago", s.Optional),
+				},
+			},
+			"slave": s.Object{
+				"offset":      c.Int("slave_repl_offset", s.Optional),
+				"priority":    c.Int("slave_priority", s.Optional),
+				"is_readonly": c.Bool("slave_read_only", s.Optional),
+			},
+			// ToDo find a way to add dynamic object of slaves: "slaves": s.Str("slaveXXX")
```
[the remark this answers is not in the document]
Comment (a3dho3yn) [reply metricbeat/module/redis/info/data.go]: Here and in some other scenarios, I need a dynamic array in the schema. Is there any way to do so? Something like this JSON:
```json
{
  "slaves": [
      {
          "host": "localhost:6380",
          "lag": 2
      },
      {
          "host": "localhost:6381",
          "lag": 0
      },   
  ]
}
```

[the remark this answers is not in the document]
Comment (jsoriano) [reply metricbeat/module/redis/info/data.go]: This kind of structure can be problematic metric-wise, to make it useful, slaves should be a [nested object](https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html), if not the event would be stored as something like:
```
{
  "slaves": {
    "host": ["localhost:6380", "localhost:6381"]
    "lag": [2, 0],
  }  
}
```
With this it is not possible, or very difficult to make proper queries with filters or aggregations.

The problem with nested objects is that we are not using them in beats and they would require some additional support, as well as some performance tests. Kibana would also need better support for queries and visualizations on these kind of fields.

This is why I proposed to have an additional metricset, that could have events with fields like:
```
"slave": {
    "host": "localhost:6380",
    "lag": 2
}
```

Or to have by now aggregations directly on the existing metricset.

**Comment by a3dho3yn** [comment]
I don't know why test_template (from test_base.py) fails :(

File: metricbeat/module/redis/info/_meta/fields.yml [hunk 2]
```diff
@@ -73,110 +67,219 @@
...
-          description:
+          description: >
             Eviction policy to use when memory limit is reached.
 
+        - name: fragmentation.ratio
+          type: float
+          description: >
+            Ratio between used_memory_rss and used_memory
+        - name: fragmentation.bytes
+          type: bytes
```
Comment (jsoriano) [inline metricbeat/module/redis/info/_meta/fields.yml]: @a3dho3yn I have run the failing test locally, and it seems that the failure is caused by this, `bytes` can be used as `format`, but not as `type`.

File: metricbeat/module/redis/info/_meta/fields.yml [hunk 3]
```diff
@@ -185,31 +289,82 @@
...
           type: long
-          description:
+          description: >
+            Number of connected slaves
 
         - name: master_offset
           type: long
-          description:
-
+          deprecated: true
```
Comment (jsoriano) [inline metricbeat/module/redis/info/_meta/fields.yml]: After merging #7779 today, `deprecated` needs to be set to the version the field is going to be deprecated on, in this case set it to 6.5.

**Comment by jsoriano** [comment]
@a3dho3yn run `make update` after modifying the fields

**Comment by a3dho3yn** [comment]
I've run `make update` but surprisingly it did not updated `docs` :|

**Comment by jsoriano** [comment]
Oh yes, sorry, you also need to update your branch with master.

**Comment by jsoriano** [comment]
jenkins, test this please

**Comment by jsoriano** [comment]
@a3dho3yn merged, thanks a lot for this contribution and the others you are working on :smiley: 

Just curious, are you using the provided dashboards for redis and rabbitmq? We'd be happy to accept contributions also there if you have any proposal :slightly_smiling_face:

**Comment by a3dho3yn** [comment]
Thank you very much for your help to complete these PRs.
We just started using ELK stack for monitoring our applications and databases. I didn't work with dashboards yet, but I think I will start working with them soon and then I will contribute if I find some deficiencies or issues :)

**Pull request merged**